File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        HashSet<int> seen = new HashSet<int>(nums);

        int longestStreak = 0;
        foreach(int n in seen){
            if(!seen.Contains(n - 1)){
                int currentNum = n;
                int currentStreak = 1;

                while(seen.Contains(currentNum + 1)){
                    currentNum += 1;
                    currentStreak += 1;
                }

                longestStreak = Math.Max(longestStreak, currentStreak);
            }
        }

        return longestStreak;
    }
}

File: Data Structures & Algorithms/non-cyclical-number/submission-3.java
class Solution {
    public boolean isHappy(int n) {
        HashSet<Integer> set = new HashSet<>();

        int sum = 0;

        while(n != 1){
            sum = 0;
            while(n > 0){
                sum+= (n % 10) * (n % 10);
                n /= 10;
            }

            if(!set.contains(sum)){
                set.add(sum);
            }else{
                return false;
            }

            n = sum;
        }

        return true;
    }
}
